### PR TITLE
[ML] Force stop deployment in use

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Implements a `force` parameter to the stop deployment API.
This allows a user to forcefully stop a deployment. Currently,
this specifically allows stopping a deployment that is in use
by ingest processors.

### CODE REVIEW

**Review by davidkyle — APPROVED** [review]
LGTM

File: docs/reference/ml/df-analytics/apis/stop-trained-model-deployment.asciidoc [hunk 1]
```diff
@@ -31,10 +31,17 @@ experimental::[]
...
 == {api-query-parms-title}
-////
+
+`allow_no_match`::
+(Optional, Boolean)
+include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=allow-no-match]
+
+
+`force`::
+  (Optional, Boolean) If true, the deployment is stopped forcefully.
```
Comment (benwtrent) [inline docs/reference/ml/df-analytics/apis/stop-trained-model-deployment.asciidoc:44]:
```suggestion
(Optional, Boolean) If true, the deployment is stopped even if it is referenced by ingest pipelines.
```

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Can we add YAML tests for these new parameters?

**Review by benwtrent — APPROVED** [review]

File: rest-api-spec/src/main/resources/rest-api-spec/api/ml.stop_trained_model_deployment.json [hunk 2]
```diff
@@ -26,6 +26,21 @@
...
+        "description":"Whether to ignore if a wildcard expression matches no deployments. (This includes `_all` string or when no deployments have been specified)"
+      },
+      "force":{
+        "type":"boolean",
+        "required":false,
+        "description":"True if the deployment should be forcefully stopped"
+      }
+    },
+    "body":{
+      "description":"The stop deployment parameters"
```
Comment (sethmlarson) [inline rest-api-spec/src/main/resources/rest-api-spec/api/ml.stop_trained_model_deployment.json:43]: What sorts of options can be passed in the `body`? I don't see a request body documented for this API.

Comment (benwtrent) [reply rest-api-spec/src/main/resources/rest-api-spec/api/ml.stop_trained_model_deployment.json:43]: There isn't one. I am not a HUGE fan of this convention (its elsewhere in ES), where the query params can optionally be supplied in the body. I would much rather have one or the other, but this change does follow convention.

**Review by sethmlarson — APPROVED** [review]
Going to switch my review to approve to not be in the way, but it would be appreciated to add docs+tests for these parameters and the request body. We currently don't have this API modeled in the [Elasticsearch specification](https://github.com/elastic/elasticsearch-specification/tree/main/specification/ml) and these resources help us accomplish that! :bowing_man:

**Comment by benwtrent** [comment]
> We currently don't have this API modeled in the Elasticsearch specification and these resources help us accomplish that!

It is experimental :D. So, specification might be a ways off...

**Comment by sethmlarson** [comment]
@benwtrent We do model experimental APIs (`knn_search`, `tasks.*`) to get support for them in language clients.

**Comment by dimitris-athanasiou** [comment]
@sethmlarson There is a problem with having a YML test for this API. To set up such a test we'd need to first start a deployment and that takes too long for actual models. We have a small dummy model we use for testing in `PyTorchModelIT`. Perhaps we can make that an asset to use in YML tests but we've avoided to do so up to now. I'll look into it and whether it makes the tests too slow.

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/bwc

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/docs

**Comment by sethmlarson** [comment]
@dimitris-athanasiou No problem! Also if only the negative cases are YAML-testable we get value from those too.

**Pull request merged**